api-client/src/i18n/DateFormatter.ts

```diff
@@ -11,25 +11,26 @@ export const ISO8601_FORMAT = undefined
 class DateFormatter {
   fallbackFormat?: string = ISO8601_FORMAT
   defaultLocale: string
-
   constructor(defaultLocale: string, fallbackFormat?: string) {
     this.fallbackFormat = fallbackFormat
     this.defaultLocale = defaultLocale
   }
 
   format: FormatFunctionType = (
-    date: Moment,
+    date: DateTime,
     options: {
       format?: string
     }
   ) => {
     const format = options.format || this.fallbackFormat
     // TODO IGAPP-399: Uncomment again and use locale instead of hardcoded 'en'
-    // const requestedLocale = defaultLocale
     const requestedLocale = 'en'
-    const allLocales = moment.locales()
-    const locale = allLocales.includes(requestedLocale) ? requestedLocale : this.defaultLocale
-    return date.locale(locale).format(format)
+    const allLocales = date.locale
+    const requestedDate = date.setLocale(requestedLocale)
+    const defaultDate = date.setLocale(this.defaultLocale)
+
+    const locale = allLocales.includes(requestedLocale) ? requestedDate : defaultDate
+    return format ? locale.toFormat(format) : locale.toISO()
```

Comment:
As far as I understand the docs, with luxon it is no longer necessary to check if the requested locale exists or not. Therefore this can be changed to 
```suggestion
    return format ? date.toFormat(format) : date.toISO()
```
Edit: I also checked it, if a locale (following the specifications) is not available, it just uses the default locale. The fix that the actual locale is used should then be done in IGAPP-399.

Reply:
@steffenkleinle I commited your suggestion